OWML.ModHelper.Interaction/ModInteraction.cs

```diff
@@ -1,36 +1,58 @@
 ﻿using OWML.Common;
-using OWML.ModHelper.Events;
 using System.Collections.Generic;
 using System.Linq;
+using UnityEngine;
 
 namespace OWML.ModHelper.Interaction
 {
     public class ModInteraction : IModInteraction
     {
-        private readonly List<ModBehaviour> _modList = new List<ModBehaviour>();
-        private readonly Dictionary<IModBehaviour, List<IModBehaviour>> _dependantDict = new Dictionary<IModBehaviour, List<IModBehaviour>>();
+        private readonly IList<ModBehaviour> _modList;
+        private Dictionary<ModBehaviour, List<ModBehaviour>> _dependantDict = new Dictionary<ModBehaviour, List<ModBehaviour>>();
+        private Dictionary<ModBehaviour, List<ModBehaviour>> _dependencyDict = new Dictionary<ModBehaviour, List<ModBehaviour>>();
 
-        public ModInteraction(List<ModBehaviour> list)
+        public ModInteraction(IList<ModBehaviour> list)
         {
             _modList = list;
 
+            RegenDicts();
+        }
+
+        private void RegenDicts()
+        {
+            _dependantDict = new Dictionary<ModBehaviour, List<ModBehaviour>>();
+            _dependencyDict = new Dictionary<ModBehaviour, List<ModBehaviour>>();
             foreach (var mod in _modList)
             {
-                var dependants = new List<IModBehaviour>();
+                var dependants = new List<ModBehaviour>();
+                var dependencies = new List<ModBehaviour>();
                 foreach (var dependency in _modList)
-                {
-                    if (dependency.ModHelper.Manifest.Dependencies != new string[] { } && dependency.ModHelper.Manifest.Dependencies.Contains(mod.ModHelper.Manifest.Name))
+                { 
+                    if (dependency.ModHelper.Manifest.Dependencies != new string[] { } && dependency.ModHelper.Manifest.Dependencies.Contains(mod.ModHelper.Manifest.UniqueName))
                     {
                         dependants.Add(dependency);
                     }
+
+                    if (mod.ModHelper.Manifest.Dependencies.Contains(dependency.ModHelper.Manifest.UniqueName))
+                    {
+                        dependencies.Add(dependency);
+                    }
                 }
                 _dependantDict.Add(mod, dependants);
+                _dependencyDict.Add(mod, dependencies);
             }
         }
 
         public IList<IModBehaviour> GetDependants(string dependencyUniqueName)
         {
-            return _dependantDict.Where(x => x.Key.ModHelper.Manifest.UniqueName == dependencyUniqueName).FirstOrDefault().Value;
+            RegenDicts();
```

Comment:
Why do you have to regenerate the dictionaries, can they change? The constructor will run before anyone can call GetDependants/GetDependencies

Reply:
because of how mod loading works in general, the ModInteraction is made before the modbehaviour. that means that in the constructor, modinteraction only has access to all the mods that loaded *before* it loaded. this should be the same for _modList, but somehow it updates?? but the dict doesn't